src/nvim/window.c

```diff
@@ -6554,6 +6549,8 @@ void win_set_inner_size(win_T *wp, bool valid_cursor)
   wp->w_width_outer = (wp->w_width_inner + win_border_width(wp));
   wp->w_winrow_off = wp->w_border_adj[0] + wp->w_winbar_height;
   wp->w_wincol_off = wp->w_border_adj[3];
+  wp->w_redr_status = true;
+  wp->w_pos_changed = true;
```

Comment:
Moving `w_redr_status` here does seem better, as `w_redr_status` is also used for redrawing winbar, which is in the inner window. However moving `w_pos_changed` here may cause unnecessary `win_pos` UI event, e.g. when adding or removing a winbar, so I don't think it should be moved here.

Reply:
Right agreed, should have done a grep on `win_set_inner_size()`. Fixed and removed the `w_redr_status` assignment from `set_winbar_win()`, which also calls `win_set_inner_size()`.